json_rpc/src/error.rs

```diff
@@ -125,7 +125,7 @@ impl Error {
     pub fn new<C: ErrorCodeT, T: Serialize>(error_code: C, additional_info: T) -> Self {
         let (code, message): (i64, &'static str) = error_code.into();
 
-        if !C::is_reserved() && code >= -32768 && code <= -32100 {
+        if !C::is_reserved() && (-32768..=-32100).contains(&code) {
```

Comment:
Are these really equivalent? They do in terms of the outcome but the old one is just 2 comparisons where the new one is an iterator over couple hundred elements of a list.

Reply:
It's equal as `(x..=y)` is a `Range` object where `start` and `end` is known, so `contains` is the same logical expression. For reference, this is `Range::contains` https://doc.rust-lang.org/src/core/ops/range.rs.html#119-122, and it calls https://doc.rust-lang.org/src/core/ops/range.rs.html#810-813. It looks a bit fancier, but it's the same.